pkg/volume/glusterfs/glusterfs.go

```diff
@@ -383,19 +383,26 @@ func (b *glusterfsMounter) setUpAtInternal(dir string) error {
 
 	}
 
-	//Add backup-volfile-servers and auto_unmount options.
-	options = append(options, "backup-volfile-servers="+dstrings.Join(addrlist[:], ":"))
-	options = append(options, "auto_unmount")
+	if (len(addrlist) > 0) && (addrlist[0] != "") {
```

Comment:
Can we add unit tests for this change? We should make it in master first, then include it in this cherry-pick.

Reply:
@msau42 Sure. Let me add some unit-tests on master first. Looks like it's missing in original PR.